app/modules/users/resources.py

```diff
@@ -139,8 +139,8 @@ def patch(self, args, user):
         context = api.commit_or_abort(
             db.session, default_error_message='Failed to update user details.'
         )
+        parameters.PatchUserDetailsParameters.perform_patch(args, user)
```

Comment:
This is what I did too but after what JP said on wednesday, I think this means that if the patch fails, it doesn't rollback. 
I think this will be a topic at he meeting today

Reply:
The alternative I had was add `subtransaction=True` to all the transactions in creating submissions... I don't know what that actually does when things fail though

Reply:
@bluemellophone What are we supposed to do about this?